File: RepoTransBench/source_projects/Java/yujiasun_Distributed-Kit/src/test/java/com/distributed/limit/redis/LimitRulePublicTest.java
package com.distributed.limit.redis;

import org.junit.Test;
import static org.junit.Assert.*;

public class LimitRulePublicTest {

    @Test
    public void testGettersAndSettersPublic() {
        LimitRule rule = new LimitRule();
        rule.setSeconds(25);
        rule.setLimitCount(12);
        rule.setLockCount(5);
        rule.setLockTime(200);

        assertEquals(25, rule.getSeconds());
        assertEquals(12, rule.getLimitCount());
        assertEquals(5, rule.getLockCount());
        assertEquals(200, rule.getLockTime());
    }

    @Test
    public void testEnableLimitLockFalseWhenZeroPublic() {
        LimitRule rule = new LimitRule();
        rule.setLockTime(0);
        rule.setLockCount(0);

        assertFalse(rule.enableLimitLock());
    }

    @Test
    public void testEnableLimitLockFalseWhenOneZeroPublic() {
        LimitRule rule = new LimitRule();
        rule.setLockTime(8);
        rule.setLockCount(0);

        assertFalse(rule.enableLimitLock());

        rule.setLockTime(0);
        rule.setLockCount(6);

        assertFalse(rule.enableLimitLock());
    }

    @Test
    public void testEnableLimitLockTruePublic() {
        LimitRule rule = new LimitRule();
        rule.setLockTime(15);
        rule.setLockCount(4);

        assertTrue(rule.enableLimitLock());
    }
}

File: RepoTransBench/target_projects/Java/Go/yujiasun_Distributed-Kit/public_tests/limit_rule_public_test.go
package public_tests

import (
	"testing"

	"github.com/stretchr/testify/assert"
)

type LimitRule struct {
	limitCount int
	seconds    int
	lockCount  int
	lockTime   int
}

func (lr *LimitRule) SetSeconds(s int)      { lr.seconds = s }
func (lr *LimitRule) SetLimitCount(l int)   { lr.limitCount = l }
func (lr *LimitRule) SetLockCount(l int)    { lr.lockCount = l }
func (lr *LimitRule) SetLockTime(l int)     { lr.lockTime = l }
func (lr *LimitRule) GetSeconds() int       { return lr.seconds }
func (lr *LimitRule) GetLimitCount() int    { return lr.limitCount }
func (lr *LimitRule) GetLockCount() int     { return lr.lockCount }
func (lr *LimitRule) GetLockTime() int      { return lr.lockTime }
func (lr *LimitRule) EnableLimitLock() bool { return lr.lockTime > 0 && lr.lockCount > 0 }

func TestGettersAndSettersPublic(t *testing.T) {
	rule := &LimitRule{}
	rule.SetSeconds(25)
	rule.SetLimitCount(12)
	rule.SetLockCount(5)
	rule.SetLockTime(200)

	assert.Equal(t, 25, rule.GetSeconds())
	assert.Equal(t, 12, rule.GetLimitCount())
	assert.Equal(t, 5, rule.GetLockCount())
	assert.Equal(t, 200, rule.GetLockTime())
}

func TestEnableLimitLockFalseWhenZeroPublic(t *testing.T) {
	rule := &LimitRule{}
	rule.SetLockTime(0)
	rule.SetLockCount(0)

	assert.False(t, rule.EnableLimitLock())
}

func TestEnableLimitLockFalseWhenOneZeroPublic(t *testing.T) {
	rule := &LimitRule{}
	rule.SetLockTime(8)
	rule.SetLockCount(0)

	assert.False(t, rule.EnableLimitLock())

	rule.SetLockTime(0)
	rule.SetLockCount(6)

	assert.False(t, rule.EnableLimitLock())
}

func TestEnableLimitLockTruePublic(t *testing.T) {
	rule := &LimitRule{}
	rule.SetLockTime(15)
	rule.SetLockCount(4)

	assert.True(t, rule.EnableLimitLock())
}